src/plugins/navigation_embeddable/public/components/navigation_embeddable_panel_editor.tsx

```diff
@@ -47,38 +43,48 @@ import { NavigationEmbeddablePanelEditorLink } from './navigation_embeddable_pan
 
 import noLinksIllustrationDark from '../assets/empty_links_dark.svg';
 import noLinksIllustrationLight from '../assets/empty_links_light.svg';
-
 import './navigation_embeddable.scss';
 
 const NavigationEmbeddablePanelEditor = ({
-  onSave,
+  onSaveToLibrary,
+  onAddToDashboard,
   onClose,
-  initialInput,
+  initialLinks,
   parentDashboard,
+  isByReference,
 }: {
+  onSaveToLibrary: (newLinks: NavigationEmbeddableLink[]) => void;
```

Comment:
For type safety, so the `await` on line `239` will no longer get flagged by Typescript: 

```suggestion
  onSaveToLibrary: (newLinks: NavigationEmbeddableLink[]) => Promise<void>;
```